app/lib/event_store/event.rb

```diff
@@ -64,6 +66,35 @@ def load(str)
 
     scope :stale, -> { where.has { created_at <= TTL.ago } }
 
+    class EventRollbackError < StandardError
+      include Bugsnag::MetaData
+
+      def initialize(error:, event:)
+        event_details = event.to_h.as_json
+        self.bugsnag_meta_data = { event: event_details }
+
+        super "Error raised trying to roll back #{event}: #{self}, event details: #{event_details}"
+        set_backtrace error.backtrace
```

Comment:
```suggestion
        super "Error raised trying to roll back #{event}: #{error}, event details: #{event_details}"
        set_backtrace error.backtrace
```

I like making this reporting more useful. The canonical way to link an error to another one though is by setting `cause`. Because this is not super common in Ruby as it is in some traditional languages (like java), some tools are not super good at reporting these nested errors.

But on command line, that should be fine. Also, in case bugsnag is not good with that, we can add the actual error to the metadata.

Current implementation can be confusing.

Reply:
Yeah, I was trying to pass "cause" somehow into the exception constructor, but couldn't find a way :grimacing: `set_backtrace` was as far as I went.

I'll apply your suggestion, but is there anything else you think can be improved?

Reply:
continue in other comments